plugin/pkg/admission/podnodeselector/admission.go

```diff
@@ -125,7 +125,7 @@ func (p *podNodeSelector) Admit(a admission.Attributes) error {
 	// second selector wins
 	podNodeSelectorLabels := labels.Merge(namespaceNodeSelector, pod.Spec.NodeSelector)
 	pod.Spec.NodeSelector = map[string]string(podNodeSelectorLabels)
-	return p.Validate(a)
```

Comment:
this is intentional... the validation checks must pass to be admitted, and must still pass once all mutation is complete